### ISSUE TITLE
Updating transgrp fails due to existing directory

### ISSUE THREAD

**Issue by mtorpey** [gap-packages/PackageManager#122]
See a recent extract from `InstallAllPackages`:

```
Package "transgrp" version 3.6.3 is installed, but 3.6.5 is available. Install it? [y/N] y
#I  Retrieving PackageInfo.g from https://www.math.colostate.edu/~hulpke/transgrp/PackageInfo.g ...
#I  transgrp version 3.6.5 will be installed, replacing 3.6.3
#I  Downloading archive from URL https://www.math.colostate.edu/~hulpke/transgrp/transgrp3.6.5.tar.gz ...
#I  Saved archive to /tmp/gaptempdirwDpmIv/transgrp3.6.5.tar.gz.pkgman
#I  Package already installed at target location
#I  Target directory /home/mtorpey/.gap/pkg/transgrp exists and is non-empty
false
```

The package manager is designed with the assumption that the directories in the archives will be called `transgrp-3.6.3/` and `transgrp-3.6.5`, so the new one can be installed before the old is deleted.  Since transgrp packages are just called `transgrp/`, this fails.

We should have some check for this.

**Comment by olexandr-konovalov** [comment]
I wonder how many packages still have non-versioned directories - perhaps @fingolfin knows?

**Comment by mtorpey** [comment]
Attempted fix in 8368007.  If the destination directory already exists, it is renamed with `.old` while installation happens.

**Comment by mtorpey** [comment]
Passing, so I'll close this!
